aml/org.testeditor.aml.dsl/src/org/testeditor/aml/dsl/validation/AmlValidator.xtend

```diff
@@ -154,40 +157,54 @@ class AmlValidator extends AbstractAmlValidator {
 		]
-		if (elementHasNoStrategy && !interactionsExpectingButWithoutStrategy.empty) {
-			val message = '''Element has interactions ('«interactionsExpectingButWithoutStrategy.map[name].join(', ')»') that require a locator strategy, but none is given.'''
-			error(message, COMPONENT_ELEMENT__LOCATOR_STRATEGY, COMPONENT_ELEMENT__LOCATOR_STRATEGY__MISSING)
-		}
-	}
+		if (elementHasNoStrategy &&
+			!interactionsExpectingButWithoutStrategy.
+				empty) {
+					val message = '''Element has interactions ('«interactionsExpectingButWithoutStrategy.map[name].join(', ')»') that require a locator strategy, but none is given.'''
+					error(message, COMPONENT_ELEMENT__LOCATOR_STRATEGY, COMPONENT_ELEMENT__LOCATOR_STRATEGY__MISSING)
+				}
+			}
 
-	override checkImports(XImportSection importSection) {
-		// ignore for now
-	}
+			override checkImports(XImportSection importSection) {
+				// ignore for now
+			}
 
-	/** get a set of strings that occur more than once in the list passed */
-	private def Set<String> getStringsOccuringMoreThanOnce(Iterable<String> list) {
-		val stringsUnusedYet = list.toSet
-		val stringsOccuringMoreThanOnce = list.dropWhile [
-			val stringWasUnusedUpToNow = stringsUnusedYet.contains(it)
-			if (stringWasUnusedUpToNow) {
-				stringsUnusedYet.remove(it) // now used, remove from unused => if used a second time it is not dropped
+			@Check
+			def void checkInteractionNameIsUnique(AmlModel amlModel) {
+				val Map<String, InteractionType> interactionNames = newHashMap
+
+				amlModel.interactionTypes.forEach [
+					if (interactionNames.containsKey(it.name)) {
+						val message = MessageFormat.format(Validation_InteractionType_Name_Dublicate, it.name)
+						error(message, interactionNames.get(it.name),
+							interactionNames.get(it.name).eContainer.eContainingFeature, INTERACTION_NAME_DUPLICATION)
+						error(message, it, it.eContainer.eContainingFeature, INTERACTION_NAME_DUPLICATION)
+					} else {
+						interactionNames.put(it.name, it)
+					}
+				]
 			}
-			return stringWasUnusedUpToNow
-		]
-		return stringsOccuringMoreThanOnce.toSet
-	}
 
-	@Check
-	def void checkInteractionNameIsUnique(AmlModel amlModel) {
-		amlModel.interactionTypes => [
-			val doubleUsedInteractionNames = map[name].getStringsOccuringMoreThanOnce
-			indexed.filter[doubleUsedInteractionNames.contains(value.name)].forEach [
-				val message = '''Interaction has name ('«value.name»') which is used at least twice.'''
-				error(message, value.eContainer, value.eContainingFeature, key, INTERACTION_NAME_DUPLICATION)
-			]
-		]
-	}
-	
+			@Check
+			def void checkDupplicateTemplatesInComponentType(ElementTypeWithInteractions componentType) {
+				val Map<String, InteractionType> mapTemplateCode2InteractionType = newHashMap
+
+				componentType.interactionTypes.forEach [
+					var tempalteCode = it.template.contents.filter(TemplateText).map[value].join
```

Comment:
within TclModelUtil the matching between interaction and template is done (line 113 getInteraction). The method to generate a string signature of the template is 'normalize'. We should use this here, too!